src/commands/createNewProject/validateFuncCliInstalled.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as opn from 'opn';
+import { MessageItem, OutputChannel } from 'vscode';
+import { DialogResponses, IAzureUserInput } from 'vscode-azureextensionui';
+import { Platform } from '../../constants';
+import { localize } from '../../localize';
+import { updateGlobalSetting } from '../../ProjectSettings';
+import { cpUtils } from '../../utils/cpUtils';
+import { functionRuntimeUtils } from '../../utils/functionRuntimeUtils';
+
+export async function validateFuncCliInstalled(ui: IAzureUserInput, outputChannel: OutputChannel): Promise<void> {
+    const runtime: string | null = await functionRuntimeUtils.getLocalFunctionRuntimeVersion();
+    const settingKey: string = 'showFuncInstallation';
+    if (runtime === null && process.platform !== Platform.Linux) {
+        // command currently does not support Linux
+        const input: MessageItem = await ui.showWarningMessage(localize('installFuncCli', 'You need Azure Functions Core Tools CLI to locally debug your functions.  Run {0} now?', getCommandForPlatform()), DialogResponses.yes, DialogResponses.dontWarnAgain, DialogResponses.skipForNow);
+        if (input === DialogResponses.yes) {
+            attemptToInstallLatestFunctionRuntime(ui, outputChannel);
+        } else if (input === DialogResponses.dontWarnAgain) {
+            await updateGlobalSetting(settingKey, false);
+        }
+    }
+}
+
+async function attemptToInstallLatestFunctionRuntime(ui: IAzureUserInput, outputChannel: OutputChannel): Promise<void> {
+    try {
+        switch (process.platform) {
+            case Platform.Windows:
+                await cpUtils.executeCommand(outputChannel, undefined, 'npm', 'install', '-g', 'azure-functions-core-tools@core');
+                break;
+            case Platform.MacOS:
+                await cpUtils.executeCommand(outputChannel, undefined, 'brew', 'tap', 'azure/functions');
```

Comment:
I haven't had any success on my machine using npm due to access errors.  I'm not sure if this is a common issue or not (since mine is straight out of the box)

Reply:
Up until very recently, npm was the only way to install the func cli on a mac. I'm guessing the issue is somewhat unique to your machine

Reply:
Okay, I can add it back in since it doesn't hurt.